### ISSUE TITLE
-AwarnUnneededSuppressions is a silent no-op under -AassumeInitialized

### ISSUE THREAD

**Issue by aosen-xiong** [eisop/checker-framework#2029]
`-AwarnUnneededSuppressions` reports nothing when combined with `-AassumeInitialized`, with no diagnostic indicating that the check was skipped. A user running both flags is told that every `@SuppressWarnings` is needed.

This affects any subclass of `InitializationChecker`, including the Nullness Checker.

### Reproducer

```java
import org.checkerframework.checker.nullness.qual.Nullable;

class Issue {

    int needed(@Nullable String s) {
        @SuppressWarnings("nullness") // NEEDED: suppresses dereference.of.nullable
        int n = s.length();
        return n;
    }

    @SuppressWarnings("nullness") // UNNEEDED: there is nothing to suppress
    int useless() {
        return 1;
    }

    int reported(@Nullable String s) {
        return s.length(); // must still be reported
    }
}
```

Without `-AassumeInitialized`, the unneeded suppression is reported:

```
$ javac -processor nullness -AwarnUnneededSuppressions Issue.java
Issue.java:11: warning: [unneeded.suppression] warning suppression "nullness" is not used by NullnessChecker
    @SuppressWarnings("nullness") // UNNEEDED: there is nothing to suppress
    ^
Issue.java:17: error: [dereference.of.nullable] dereference of possibly-null reference s
        return s.length(); // must still be reported
               ^
1 error
1 warning
```

With `-AassumeInitialized`, it silently disappears:

```
$ javac -processor nullness -AassumeInitialized -AwarnUnneededSuppressions Issue.java
Issue.java:17: error: [dereference.of.nullable] dereference of possibly-null reference s
        return s.length(); // must still be reported
               ^
1 error
```

Note that the needed suppression on line 6 is not reported in either run, and the `dereference.of.nullable` error on line 17 is still reported under `-AassumeInitialized`, so nullness checking is still running. Only the unneeded suppression on line 11 changes between the two runs.

### Expected behaviour

Either report unneeded suppressions collected by the subcheckers, or issue a `UserError` stating that `-AwarnUnneededSuppressions` is not supported together with `-AassumeInitialized`.

### Environment

- Checker Framework 3.49.5-eisop2-SNAPSHOT, commit `e434ef1` (unmodified `master`)
- JDK 17
